### PR TITLE
chore: move autoware interfaces

### PR DESCRIPTION
**Pull request by isamu-takagi**
Add "autoware interface" directory for documents common to "AD API" and "Component Interface".

### CODE REVIEW

File: docs/design/autoware-interfaces/component/.pages [hunk 1]
```diff
@@ -0,0 +1,2 @@
+nav:
+  - index.md
```
Comment (kenji-miyake) [inline docs/design/autoware-interfaces/component/.pages:2]:
```suggestion
  - index.md
  - planning-control-vehicle-interface.md
```

**Comment by kenji-miyake** [comment]
@isamu-takagi Could you edit `.pages` so that the page hierarchy will be shown correctly?
image

When the order isn't important, you can use `...`.
https://github.com/lukasgeiter/mkdocs-awesome-pages-plugin#rest

**Comment by kenji-miyake** [comment]
After 6d20e14:
image

File: docs/design/autoware-interfaces/index.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Autoware Interfaces
```
Comment (kenji-miyake) [inline docs/design/autoware-interfaces/index.md:1]: Please use `sentence case`.
https://developers.google.com/style/headings?hl=en#heading-and-title-format

```suggestion
# Autoware interfaces
```

File: docs/design/autoware-interfaces/component/.pages [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+  - Architecture: planning-control-vehicle-interface.md
```
Comment (kenji-miyake) [inline docs/design/autoware-interfaces/component/.pages:3]: Why did you customize the title here?

Comment (isamu-takagi) [reply docs/design/autoware-interfaces/component/.pages:3]: Because it is too long for menu.

Comment (kenji-miyake) [reply docs/design/autoware-interfaces/component/.pages:3]: I see, but writing just `Architecture` is ambiguous to me. :thinking:

Comment (kenji-miyake) [reply docs/design/autoware-interfaces/component/.pages:3]: So how about splitting this file into `planning.md`, `control.md`, and `vehicle-interface.md`?

cc @TakaHoribe

Comment (kenji-miyake) [reply docs/design/autoware-interfaces/component/.pages:3]: Also, creating empty files for other components would be good.

Comment (kenji-miyake) [reply docs/design/autoware-interfaces/component/.pages:3]: @isamu-takagi cc @TakaHoribe ping.

Comment (isamu-takagi) [reply docs/design/autoware-interfaces/component/.pages]: @kenji-miyake @TakaHoribe 
Split the component architecture page. How about this? https://github.com/autowarefoundation/autoware-documentation/pull/57/commits/0308758a166568dd6f93cbee0e474e1cab82b72a

**Review by kenji-miyake — DISMISSED** [review]
Looks cool! Thank you. :+1:

File: docs/design/autoware-interfaces/.pages [hunk 4]
```diff
@@ -0,0 +1,4 @@
+nav:
+  - index.md
+  - AD API: ad-api
+  - Components: components
```
Comment (kenji-miyake) [inline docs/design/autoware-interfaces/.pages:4]: [Nits] Unnecessary?

```suggestion
  - components
```

Comment (isamu-takagi) [reply docs/design/autoware-interfaces/.pages:4]: Ah, that's right. I confused what is used for menu text. Directory name is used for directory, and page title for markdown.

Comment (isamu-takagi) [reply docs/design/autoware-interfaces/.pages]: https://github.com/autowarefoundation/autoware-documentation/pull/57/commits/3b1dade01f4c3d124ed704f949589c99222d13df

**Comment by kenji-miyake** [comment]
cc @xmfcx We've cleaned up the Bus ODD's component architecture pages.
image

**Review by kenji-miyake — APPROVED** [review]

**Pull request merged**